File: Projects/EspressoDemos/EspressoEffectiveTesting/app/src/test/java/com/example/code/data/RecipeTest.java
package com.example.code.data;

import org.junit.Test;

import java.io.InputStream;

import static org.junit.Assert.*;

public class RecipeTest {

    @Test
    public void readFromStream() {
        InputStream stream = RecipeTest.class.getResourceAsStream("/recipes/water.txt");
        Recipe recipe = Recipe.readFromStream(stream);
        assertNotNull(recipe);
    }

}

File: Projects/RecepieApp/app/src/test/java/com/example/effectivetesting/model/RecipeTest.kt
package com.example.effectivetesting.model

import com.example.effectivetesting.model.Recipe.staticMethods.readFromStream
import org.junit.Assert.*
import org.junit.Test
import java.io.InputStream

class RecipeTest {

    @Test
    fun `Check if reading of the asset file returns a null value`() {
        val stream = RecipeTest::class.java.getResourceAsStream("/recipes/water.txt")

        readFromStream(stream)?.let {
            val recipe: Recipe = it
            assertNotNull(recipe)
            assertEquals("water", recipe.id)
            assertEquals("Water", recipe.title)
            assertEquals("Put glass under tap. Open tap. Close tap. Drink.", recipe.description)
        }

    }


}